clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/security/EcsCredentialsInitializer.java

```diff
@@ -40,55 +47,50 @@ public ECSCredentialsConfig ecsCredentialsConfig() {
   }
 
   @Bean
-  @DependsOn("netflixAmazonCredentials")
-  public List<? extends NetflixAmazonCredentials> netflixECSCredentials(
-      CredentialsLoader<? extends NetflixAmazonCredentials> credentialsLoader,
-      ECSCredentialsConfig credentialsConfig,
-      AccountCredentialsRepository accountCredentialsRepository)
-      throws Throwable {
-    return synchronizeECSAccounts(
-        credentialsLoader, credentialsConfig, accountCredentialsRepository);
+  @DependsOn("amazonCredentialsLoader")
+  @ConditionalOnMissingBean(
```

Comment:
Yes, `ConditionalOnMissingBean` should still be here to allow plugins to create custom repository beans. There is an unintended effect of using `parameterizedContainer` though. Using it with `value = NetflixECSCredentials.class` would match both `NetflixECSCredentials` AND `CredentialsRepository<NetflixECSCredentials>` beans. So it wouldn't be a problem now but if someone creates a bean of type `NetflixECSCredentials`, it would cause this to not create repository bean. It's probably a better idea to use `name` instead.